Initial support for rockchip boards (#8382)
blakeblackshear/frigate: docs/docs/configuration/object_detectors.md

# Officially Supported Detectors

> Frigate provides the following builtin detector types: `cpu`, `edgetpu`, `openvino`, `tensorrt` and `rknn`. By default, Frigate will use a single CPU detector. Other detectors may require additional configuration as described below. When using multiple detectors they will run in dedicated processes, but pull from a common queue of detection requests from across all cameras.

Reviewer: ```suggestion
Frigate provides the following builtin detector types: `cpu`, `edgetpu`, `openvino`, `tensorrt`, and `rknn`. By default, Frigate will use a single CPU detector. Other detectors may require additional configuration as described below. When using multiple detectors they will run in dedicated processes, but pull from a common queue of detection requests from across all cameras.
```

# Community Supported Detectors

## Rockchip RKNN-Toolkit-Lite2

This detector is only available if one of the following Rockchip SoCs is used:
- RK3566/RK3568
- RK3588/RK3588S
- RV1103/RV1106
- RK3562

These SoCs come with a NPU that will highly speed up detection.

### Setup

RKNN support is provided using the `-rk` suffix for the docker image. Moreover, privileged mode must be enabled by adding the `--privileged` flag to your docker run command or `privileged: true` to your `docker-compose.yml` file.

> ## Configuration

Reviewer: ```suggestion
### Configuration
```

> # Community Supported Detectors

Reviewer: ```suggestion
# Community Supported Detectors

```

## Rockchip RKNN-Toolkit-Lite2
This detector is only available if one of the following Rockchip SoCs is used:
- RK3566/RK3568
- RK3588/RK3588S
- RV1103/RV1106
- RK3562

These SoCs come with a NPU that will highly speed up detection.

> ### Setup

Reviewer: ```suggestion
### Setup

```

All you have to do is to use the `frigate:latest-rk` docker image. Here is a minimal `docker-compose.yml`:
```
version: "3.9"
services:
  frigate:
    container_name: frigate
    image: frigate:latest-rk
    privileged: true
    restart: unless-stopped
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./data/config:/config
      - ./data/media:/media/frigate
    ports:
      - "5000:5000"
```

> ## Configuration
> This `config.yml` shows all relevant options to configure the detector and explains them. All values shown are the default values (except for one). Lines that are required at least to use the detector are labeled as required, all other lines are optional.

Reviewer: ```suggestion
## Configuration

This `config.yml` shows all relevant options to configure the detector and explains them. All values shown are the default values (except for one). Lines that are required at least to use the detector are labeled as required, all other lines are optional.
```

> ```

Reviewer: this should have `yaml` so it has syntax highlighting

# Community Supported Detectors

> ## Rockchip RKNN-Toolkit-Lite2
> This detector is only available if one of the following Rockchip SoCs is used:

Reviewer: ```suggestion
## Rockchip RKNN-Toolkit-Lite2

This detector is only available if one of the following Rockchip SoCs is used:
```

- RK3566/RK3568
- RK3588/RK3588S
- RV1103/RV1106
- RK3562

These SoCs come with a NPU that will highly speed up detection.

### Setup

> All you have to do is to use the `frigate:latest-rk` docker image. Here is a minimal `docker-compose.yml`:

Reviewer: ```suggestion
RKNN support is provided using the `-rk` suffix for the docker image.
```

```
version: "3.9"
services:
  frigate:
    container_name: frigate
    image: frigate:latest-rk
    privileged: true
    restart: unless-stopped
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - ./data/config:/config
      - ./data/media:/media/frigate
    ports:
      - "5000:5000"

> ```

Reviewer: I think we should avoid showing the docker compose here, the installation docs are used for this already. This would create a higher chance that changes to the docker compose file in the future are forgotten to be reflected here

Author: Yes, maybe I will just write, that docker must run in privileged mode.

Reviewer: Yeah makes sense
